onnxruntime/core/providers/nnapi/nnapi_builtin/builders/op_builder.cc

```diff
@@ -2843,29 +2812,13 @@ Status PadOpBuilder::AddToModelBuilderImpl(ModelBuilder& model_builder, const No
   const auto* pads_initializer = model_builder.GetConstantInitializer(pads);
   ORT_RETURN_IF_NOT(pads_initializer, "pads must be a constant");
 
-  std::vector<uint8_t> pads_initializer_raw_data{};
-  ORT_RETURN_IF_ERROR(utils::UnpackInitializerData(*pads_initializer, pads_initializer_raw_data));
+  Initializer pads_initializer_raw_data(*pads_initializer);
   // assume pads_initializer has int64 data, per ONNX spec
-  ORT_RETURN_IF_NOT(pads_initializer_raw_data.size() == 2 * data_rank * sizeof(int64_t),
-                    "Expected pads initializer size in bytes: ", 2 * data_rank * sizeof(int64_t),
-                    ", actual: ", pads_initializer_raw_data.size());
-
   std::vector<int32_t> converted_pads_data{};
   converted_pads_data.reserve(2 * data_rank);
-
-  auto copy_and_convert = [](const void* raw_i64_src,
-                             std::back_insert_iterator<decltype(converted_pads_data)> i32_dst) {
-    int64_t i64;
-    memcpy(&i64, raw_i64_src, sizeof(i64));
-    *i32_dst = SafeInt<int32_t>(i64);
-  };
-
   for (size_t i = 0; i < data_rank; ++i) {
-    copy_and_convert(&pads_initializer_raw_data[i * sizeof(int64_t)],
-                     std::back_inserter(converted_pads_data));
-
-    copy_and_convert(&pads_initializer_raw_data[(i + data_rank) * sizeof(int64_t)],
-                     std::back_inserter(converted_pads_data));
+    converted_pads_data.push_back(pads_initializer_raw_data.DataAsSpan<int64_t>()[i]);
+    converted_pads_data.push_back(pads_initializer_raw_data.DataAsSpan<int64_t>()[i + data_rank]);
```

Comment:
There's a narrowing conversion from int64 to int32 here that should have a gsl::narrow or SafeInt on it.

Whilst the optimizer can probably get rid of it, it would be better to call DataAsSpan once outside the loop (as that creates a new span object) and re-use it in the loop. This applies in a number of places.

Reply:
Good suggestion. 
Fixed